portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,81 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 import java.io.IOException;
+import java.util.ArrayList;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Returns an array of comments in JSON form */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+  int maxComments=3;
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  /* Returns an array of comments in JSON 
```

Comment:
nit: Javadoc should look a little more like https://google.github.io/styleguide/javaguide.html#s7.1.1-javadoc-multi-line. For this, it can probably just be a single line `/** Returns an array of comments in JSON with size maxComments. */`